File: alphabetSubSequence.ts
function alphabetSubsequence(inputString: String): boolean {
  const chars: string[] = inputString.split("");
  const charValues: number[] = [];
  chars.forEach((char: string) => {
    charValues.push(char.charCodeAt(0));
  });
  if (new Set(charValues).size !== charValues.length) {
    return false;
  }
  for (let i = 0; i < charValues.length; i++) {
    if (charValues[i] >= charValues[i + 1]) {
      return false;
    }
  }
  return true;
}
console.log(alphabetSubsequence("effg"));
console.log(alphabetSubsequence("cdce"));
console.log(alphabetSubsequence("ace"));
console.log(alphabetSubsequence("bxz"));
console.log(alphabetSubsequence("bpez"));
console.log(alphabetSubsequence("bza"));

File: AlphabetSubSequence.java
public class AlphabetSubSequence {

    public static void main(String[] args){

        System.out.println(alphabetSubSequence("vigny"));
        System.out.println(alphabetSubSequence("ace"));
        System.out.println(alphabetSubSequence("bxz"));
        System.out.println(alphabetSubSequence("effg"));
    }
    public static boolean alphabetSubSequence(String s){
        String alphabet = "abcdefghijklmnopqrstuvwxyz";

        char[] characters = s.toCharArray();
        int[] carValues = new int[characters.length];
        for(int i=0;i<characters.length; i++){
            carValues[i]= (int )characters[i];
        }
        return isSortedWithoutDoubles(carValues);

        //return false;
    }

    public static boolean isSortedWithoutDoubles(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i] >= arr[i+ 1]) {
                return false;
            }
        }
        return true;
    }
}
